src/server/page_server.cpp

```diff
@@ -204,18 +204,80 @@ page_server::connection_handler::receive_start_catch_up (tran_server_conn_t::seq
   unpacker.unpack_int (port);
   cublog::lsa_utils::unpack (unpacker, catchup_lsa);
 
-  er_log_debug (ARG_FILE_LINE, "receive_start_catch_up: hostname = %s, port = %d, LSA = (%lld|%d)\n", host.c_str (), port,
-		LSA_AS_ARGS (&catchup_lsa));
+  _er_log_debug (ARG_FILE_LINE, "[CATCH-UP] Start catching up with the follower (%s:%d), until LSA = (%lld|%d)\n",
+		 host.c_str (), port,LSA_AS_ARGS (&catchup_lsa));
   if (port == -1)
     {
       return; // TODO: It means that the ATS is booting up, it will be set properly after ATS recovery is implemented.
     }
 
   // TODO: A thread will take the catch-up including establishing connection to avoid blocking ATS->PS reqeusts.
-  m_ps.connect_to_followee_page_server (std::move (host), port);
+  // All the below here will be taken up.
+  {
+    m_ps.connect_to_followee_page_server (std::move (host), port);
+
+    assert (!log_Gl.hdr.append_lsa.is_null ());
+    assert (!catchup_lsa.is_null ());
+    assert (catchup_lsa >= log_Gl.hdr.append_lsa);
+
+    if (log_Gl.hdr.append_lsa == catchup_lsa)
+      {
+	_er_log_debug (ARG_FILE_LINE, "[CATCH-UP] There is nothing to catch up.\n");
+	return; // TODO the cold-start case. No need to catch up.
+      }
+
+    constexpr int MAX_PAGE_REQUEST_CNT_AT_ONCE = 10; // Arbitrarily chosen
+    const LOG_PAGEID start_pageid = log_Gl.hdr.append_lsa.pageid;
+    const LOG_PAGEID end_pageid = catchup_lsa.offset == 0 ? catchup_lsa.pageid - 1 : catchup_lsa.pageid;
```

Comment:
@wrlawodms (post-merge comment)

`catchup_lsa.offset == 0`
please explain
is it because, if offset is 0 this means we also need the previous page to - possibly link wit the subsequent one?